### DISCUSSION TITLE
I miss "Check Missing Grid" and "Check missing Continent" after upgrading from Cloudlog

### DISCUSSION THREAD

**Discussion by m0urs** [wavelog/wavelog#522] [Q&A]
In Cloudlog there were two URLs which you could call from a cron job to add missing locator data and missing continent data to QSOs. This was very valuable e.g. if you imported QSOs from other logging programs (e.g. N1MM) which did not include that information. So the missing information was updated automatically from QRZ.com.

The URLs were

curl --silent https://cloudlog.acme.com/update/check_missing_continent
and
curl --silent https://cloudlog.acme/update/check_missing_grid

Both do not seem to work in Wavelog. So the question is: Is that not (yet?) implemented in Wavelog? Or do I need to use different URLs in Wavelog for these functions?

It seems that these URLs DO work, while I call them via a browser session where I am logged into Wavelog but do NOT work while called via Curl from a cron job (where there is no opened session).

If I remember correctly this was the same problem we had in Cloudlog some time ago but this has been fixed somehow.

**Comment by HB9HIL** [comment]
Hello @m0urs 

You are right. We closed that down to only available when logged in as admin. The reason is that this affects ALL qso's in the database of ALL users. This shouldn't be unprotected.
If you are logged in as admin you can find this at ADMIN -> Update Country Files

Maybe we add this functions to the cronmanager and protect them some other way.

**Reply by m0urs** [reply]
> If you are logged in as admin you can find this at ADMIN -> Update Country Files

Is that also true for updating the GRID (Locator) information? Would be far more important for me than the continent data.

Maybe it would be possible to use an R/W API key for this so it would be just as protected as e.g. the backup function?

**Reply by HB9HIL** [reply]
I got this in my backlog. Currently I'm doing a big "restructuring" of the functions (see #339). After this we have some more scenarios which we can realize with the cronmanager (calling function closed in php). Hang tight :)
